### ISSUE TITLE
Long Game Duration with Few Bots Remainin

### ISSUE THREAD

**Issue by Kurisudes** [freehuntx/gpn-tron#18]
Hi there!

First off, I want to say how much I love this game. I first discovered it at the last GPN while trolling with @freehuntx in the kitchen, and it completely captivated me. It's truly an amazing piece of work—thank you so much for creating it!

Now, onto the issue:

I've noticed that when the game starts with a large number of bots, and it eventually narrows down to just two bots, the game can drag on for a long time. This is because these two bots end up with a huge amount of space to cover.

To address this, I was wondering if it might be possible to dynamically reduce the map size whenever a bot dies. The idea would be to shrink the map so that all remaining bots' current positions still fit within the map, with maybe an additional 1-2 fields of padding. This way, the game could maintain its intensity even when only a few bots are left.

On a related note, if @freehuntx is okay with it, I'd love to host this game at the 38c3. I heard you might not be attending, but if there's anything I can do to support either this issue or the hosting, please let me know! I'd even be happy to give fixing this issue a try myself, but if anyone else wants to collaborate, that would be fantastic.

Thanks again for this incredible game and all the effort you've put into it. I'm excited to see where it goes from here!

**Comment by freehuntx** [comment]
Thanks for your awesome feedback!
I also thought about this idea, but i truely believe the game should be as fair as possible.
And i have the feeling, shrinking the map could make one side end up in a worse position (cutting ways off).
My solution to the huge space was to increase the tickrate (which bots should be able to handle, otherwise we get into a metagame of "how performant is your code and how fast can you answer?).

Maybe we should adjust the tickrate increase logic instead?

No i was never at the congress ^^ Its somehow too much for me (i love keeping it local).
Feel free to host the game wherever you want :) I maybe cannot help much but if you have questions feel free!

**Comment by Kurisudes** [comment]
Hi @freehuntx 

Thanks for the response! I get your point about reducing map size introducing more luck—it could create unfair situations by cutting off paths. That said, luck is already a factor, as it’s impossible to predict which bot will die next. This unpredictability keeps the game interesting and avoids a perfect, overly predictable solution.

As for increasing the tickrate, I think it could also be problematic. It shifts the challenge to factors like PC performance and network latency, which could make the game less enjoyable. If we wanted to test code performance, a RoboCode-style setup (compiling bots and simulating on a server) would make more sense. But that would completely change the charm of this game.

I am already trying to work on the change also to get more familiar with the architecture. (I am embedded developer. I have not much knowledge about this kind of stuff) The yellow line is the reduced limit. I also changed a few smaller things like the color of the bot names.
![Screenshot 2024-11-17 at 09 47 21](https://github.com/user-attachments/assets/c5522b4c-71c7-438c-a754-794d90da3b70)

I’ll definitely host this at 38c3 and reach out if I have questions. Thanks again for creating such a great game!

**Comment by freehuntx** [comment]
Tbh the intention of this change (tickrate increase) was actually exactly this :D So people cannot just throw some huge AI model onto this challenge (which is slower than proper code).
So late ingame, just the best of the best should survive. And in such a state you need another factor to proof "this bot is better".
And i like performance being this factor.

Maybe we have different opinions about that but thats fine for me :)
I could imagine this to be settings which can be turned on/off. (tickrate increase, map size reduce etc.)

**Comment by Kurisudes** [comment]
Yes, that is fully right. I will make it a configurable option. 

If you have any other request which I should consider or general hint, then let me know.
